protocol/daemons/liquidation/client/sub_task_runner.go

```diff
@@ -82,39 +167,120 @@ func (c *Client) GetLiquidatableSubaccountIds(
 		metrics.Latency,
 	)
 
-	// Filter out subaccounts with no open positions.
-	subaccountsToCheck := make([]satypes.SubaccountId, 0)
+	numSubaccountsWithOpenPositions := 0
+	liquidatableSubaccountIds = make([]satypes.SubaccountId, 0)
 	for _, subaccount := range subaccounts {
-		if len(subaccount.PerpetualPositions) > 0 {
-			subaccountsToCheck = append(subaccountsToCheck, *subaccount.Id)
+		// Skip subaccounts with no open positions.
```

Comment:
For my understanding, why did we add this again?

Reply:
just to early return since MMR will be 0 and subaccount won't be liquidatable. no need to calculate TNC/MMR in this case.